isis/scripts/isisVarInit.py

```diff
@@ -0,0 +1,146 @@
+# For more details about the LICENSE terms and the AUTHORS, you will
+# find files of those names at the top level of this repository.
+#
+# SPDX-License-Identifier: CC0-1.0
+
+
+def mkdir(p: Path):
+    # This just wraps and reports on the directory creation:
+    if p.exists():
+        print(f"{p} exists, don't need to create.")
+    else:
+        p.mkdir(parents=False)
+        print(f"Created {p}")
+    return
+
+
+def activate_text(shell: dict, env_vars: dict) -> str:
+    """Returns the formatted text to write to the activation script
+    based on the passed dictionaries."""
+
+    lines = [shell["shebang"]]
+    for k, v in env_vars.items():
+        lines.append(shell["activate"].format(k, v))
+
+    if shell["activate_extra"] != "":
+        lines.append(shell["activate_extra"])
+
+    lines.append("cat $ISISROOT/version")
+
+    return "\n".join(lines)
+
+
+def deactivate_text(shell: dict, env_vars: dict) -> str:
+    """Returns the formatted text to write to the deactivation script
+    based on the passed dictionaries."""
+
+    lines = [shell["shebang"]]
+    for k in env_vars.keys():
+        lines.append(shell["deactivate"].format(k))
+
+    return "\n".join(lines)
+
+
+# Set up and then parse the command line:
+parser = argparse.ArgumentParser(description=__doc__)
+
+parser.add_argument(
+    "-d",
+    "--data-dir",
+    default=os.environ["CONDA_PREFIX"] + "/data",
+    type=Path,
+    help="ISIS Data Directory, default: %(default)s",
+)
+parser.add_argument(
+    "-t",
+    "--test-dir",
+    default=os.environ["CONDA_PREFIX"] + "/testData",
+    type=Path,
+    help="ISIS Test Data Directory, default: %(default)s",
+)
+
+parser.add_argument(
+    "-a",
+    "--ale-dir",
+    default=os.environ["CONDA_PREFIX"] + "/aleData",
+    type=Path,
+    help="ISIS Ale Data Directory, default: %(default)s",
+)
+args = parser.parse_args()
+
+print("-- ISIS Data Directories --")
+# Create the data directories:
+mkdir(args.data_dir)
+mkdir(args.test_dir)
+mkdir(args.ale_dir)
+
+print("-- Conda activation and deactivation scripts --")
+# Create the conda activation and deactivation directories:
+activate_dir = Path(os.environ["CONDA_PREFIX"]) / "etc/conda/activate.d"
+deactivate_dir = Path(os.environ["CONDA_PREFIX"]) / "etc/conda/deactivate.d"
+
+mkdir(activate_dir)
+mkdir(deactivate_dir)
+
+# Set the environment variables to manage
+env_vars = dict(
+    ISISROOT=os.environ["CONDA_PREFIX"],
+    ISISDATA=args.data_dir,
+    ISISTESTDATA=args.test_dir,
+    ALESPICEROOT=args.ale_dir
+)
+
+# These dicts define the unique aspects of the shell languages
+# for setting and unsetting the environment variables.
+
+sh = dict(  # this covers bash and zsh
+    extension=".sh",
+    shebang="#!/usr/bin/env sh",
+    activate="export {}={}",
+    activate_extra="",
+    deactivate="unset {}"
+)
+
+csh = dict(
+    extension=".csh",
+    shebang="#!/usr/bin/env csh",
+    activate="setenv {}={}",
```

Comment:
setenv doesn't use an =, this should be `setenv {} {}`

Reply:
Thanks, I did mention that I didn't test it.